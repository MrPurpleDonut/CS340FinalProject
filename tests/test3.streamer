bool val1 = true;
bool val2 = false;
bool val3 = false;

bool result = val1 || val2 && !val3;
