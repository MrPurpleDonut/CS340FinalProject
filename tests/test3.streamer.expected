Program parsed correctly!
---------------------------
--------Printing AST-------
---------------------------
Program
    Function List
    Statement List
        Declaration statement: val1
            Type: bool
            Assignment statement: val1
                Boolean literal: true
        Declaration statement: val2
            Type: bool
            Assignment statement: val2
                Boolean literal: false
        Declaration statement: val3
            Type: bool
            Assignment statement: val3
                Boolean literal: false
        Declaration statement: result
            Type: bool
            Assignment statement: result
                Binary operator '||'
                    Identifier: val1
                    Binary operator '&&'
                        Identifier: val2
                        Not expression
                            Identifier: val3
